### PR TITLE
Update neode

### PR DESCRIPTION
**Pull request by Tirokk**
> [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-12-17T10:32:53Z" title="Tuesday, December 17th 2019, 11:32:53 am +01:00">Dec 17, 2019</time>_
_Merged <time datetime="2019-12-17T16:20:04Z" title="Tuesday, December 17th 2019, 5:20:04 pm +01:00">Dec 17, 2019</time>_
---

## 🍰 Pullrequest

    Update neode

    This fixes some API changes with the update of neode.

See https://github.com/Human-Connection/Human-Connection/pull/1913

This also fixes a bug along the way (wrong model name). And we have to format the validation to send them back to the client.

### CODE REVIEW

**Comment by Tirokk** [comment]
> [cypress[bot]](https://github.com/apps/cypress) **Authored by [cypress[bot]](https://github.com/apps/cypress)**
_<time datetime="2019-12-17T14:46:48Z" title="Tuesday, December 17th 2019, 3:46:48 pm +01:00">Dec 17, 2019</time>_

---



<p align="center">
  <br />
  
</p>

---

### Test summary
<a href="https://dashboard.cypress.io/projects/qa7fe2/runs/3072/specs"></a> <b title="Passed">50</b>  &bull; <a href="https://dashboard.cypress.io/projects/qa7fe2/runs/3072/failures"></a> <b title="Failed">0</b>  &bull;  <b title="Pending">0</b> &bull;  <b title="Skipped">0</b>

---

### Run details
| | |
| --- | --- |
| **Project** | Human-Connection |
| **Status** | Passed |
| **Commit** | e5e600ea5a |
| **Started** | Dec 17, 2019 2:35 PM |
| **Ended** | Dec 17, 2019 2:51 PM |
| **Duration** | 15:29 <a href="https://on.cypress.io/parallelization" title="Enable parallelization to reduce run duration">💡</a> |
| **OS** | Linux Ubuntu Linux - 16.04 |
| **Browser** | Chromium 79  |

[View run in Cypress Dashboard ➡️](https://dashboard.cypress.io/projects/qa7fe2/runs/3072/specs)



---

This comment has been generated by **cypress-bot** as a result of this project's GitHub integration settings. You can manage this integration in [this project's settings in the Cypress Dashboard](https://dashboard.cypress.io/projects/qa7fe2/settings)

File: backend/src/schema/resolvers/registration.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ export default {
       `,
         { nonce, email },
       )
-      const emailAddress = await neode.hydrateFirst(result, 'email', neode.model('Email'))
+      const emailAddress = await neode.hydrateFirst(result, 'email', neode.model('EmailAddress'))
       if (!emailAddress) throw new UserInputError('Invalid email or nonce')
```
Comment (Tirokk) [inline backend/src/schema/resolvers/registration.js:44]: > [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-12-17T16:19:43Z" title="Tuesday, December 17th 2019, 5:19:43 pm +01:00">Dec 17, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -40,7 +40,7 @@ export default {
       `,
         { nonce, email },
       )
-      const emailAddress = await neode.hydrateFirst(result, 'email', neode.model('Email'))
+      const emailAddress = await neode.hydrateFirst(result, 'email', neode.model('EmailAddress'))
       if (!emailAddress) throw new UserInputError('Invalid email or nonce')
```


what would we have to do to remove `neode` from production code?
This is one of the only remaining resolvers which uses it...

Comment (Tirokk) [reply backend/src/schema/resolvers/registration.js:44]: > [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-12-17T18:03:10Z" title="Tuesday, December 17th 2019, 7:03:10 pm +01:00">Dec 17, 2019</time>_

---


The fact that @johnymontana considers using neode in neo4j-graphql-js is making me unsure about if we should try to refactor neode out of our production code.

**Review by Mogge — APPROVED** [review]
great @roschaefer

**Pull request closed without merging**